src/tools/manifest2proto.ts

```diff
@@ -336,7 +336,10 @@ export async function typeToProtoPayload(type: Type) {
     case 'TypeVariable': {
       const constraintType = type.canReadSubset || type.canWriteSuperset;
       const name = {name: (type as TypeVariable).variable.name};
-      const constraint = constraintType ? {constraint: {constraintType: await typeToProtoPayload(constraintType)}} : {};
+      const constraint = {constraint: {unconstrained: (type as TypeVariable).variable.resolveToMaxType || false}};
+      if (constraintType) {
+        constraint.constraint['constraintType'] = await typeToProtoPayload(constraintType);
```

Comment:
do we need to add a check that if `resolveToMaxType` is `true`, `constraintType` should be null?